### PR TITLE
Lists inconsistencies

### LINKED ISSUE #2544: [Android] Inconsistency with MvxRecyclerView vs. MvxListView & MvxSpinner
MvxRecyclerView, MvxListView, and MvxSpinner can all be used to show a collection to the user. All three support using an item template to allow custom cells in the list to take advantage of MvvmCross data binding. If you do not specify an item template for an MvxListView or MvxSpinner then each cell of the list will be displayed using `ToString()`. If you do not specify an item template for an MvxRecyclerView then an exception is thrown at runtime. MvxRecyclerView seems to be programmed to require a MvxItemTemplate without providing a default like MvxListView and MvxSpinner. Does it make sense to update MvxRecyclerView to work the same way as MvxSpinner and MvxListView?

### Steps to reproduce :scroll:

1. Create a MvxRecyclerView in AXML file.

2. Do not specify the MvxItemTemplate for the MvxRecyclerView.
```
<MvvmCross.Droid.Support.V7.RecyclerView.MvxRecyclerView
    android:id="@+id/code_recycler_view"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    app:MvxBind="ItemsSource Codes; ItemClick SelectCodeCommand;" />
```
3. Navigate to the view to throw Android.Content.Res.Resources+NotFoundException exception. If you replace the `<MvxRecyclerView>` XML tag with `<MvxSpinner>` or `<MvxListView>` then the list renders (using `.ToString`) and no exceptions are thrown.

### Expected behavior :thinking:
I think MvxRecyclerView should work like MvxListView and MvxSpinner and populate the list using `ToString()` by default if no item template is used.

### Actual behavior :bug:
Exception is thrown at runtime: Android.Content.Res.Resources+NotFoundException: Resource ID #0x0. This happens because the item template identifier is an int initialized to 0 then not set by my code.

```
UNHANDLED EXCEPTION:
Android.Content.Res.Resources+NotFoundException: Resource ID #0x0
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <657aa8fea4454dc898a9e5f379c58734>:0 
  at Java.Interop.JniEnvironment+InstanceMethods.CallNonvirtualObjectMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00089] in <e736913786c2475188869561ae512b72>:0 
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualObjectMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x0005d] in <e736913786c2475188869561ae512b72>:0 
  at Android.Views.LayoutInflater.Inflate (System.Int32 resource, Android.Views.ViewGroup root, System.Boolean attachToRoot) [0x0005a] in <8211f1f12f9e45838d631abf63e51c79>:0 
  at MvvmCross.Binding.Droid.Views.MvxLayoutInflater.Inflate (System.Int32 resource, Android.Views.ViewGroup root, System.Boolean attachToRoot) [0x0003e] in C:\projects\mvvmcross\MvvmCross\Binding\Droid\Views\MvxLayoutInflater.cs:137 
  at MvvmCross.Binding.Droid.BindingContext.MvxAndroidBindingContext.CommonInflate (System.Int32 resourceId, Android.Views.ViewGroup viewGroup, System.Boolean attachToRoot) [0x00013] in C:\projects\mvvmcross\MvvmCross\Binding\Droid\BindingContext\MvxAndroidBindingContext.cs:52 
  at MvvmCross.Binding.Droid.BindingContext.MvxAndroidBindingContext.BindingInflate (System.Int32 resourceId, Android.Views.ViewGroup viewGroup, System.Boolean attachToRoot) [0x00000] in C:\projects\mvvmcross\MvvmCross\Binding\Droid\BindingContext\MvxAndroidBindingContext.cs:36 
  at MvvmCross.Droid.Support.V7.RecyclerView.MvxRecyclerAdapter.InflateViewForHolder (Android.Views.ViewGroup parent, System.Int32 viewType, MvvmCross.Binding.Droid.BindingContext.IMvxAndroidBindingContext bindingContext) [0x0000d] in C:\projects\mvvmcross\MvvmCross-AndroidSupport\MvvmCross.Droid.Support.V7.RecyclerView\MvxRecyclerAdapter.cs:175 
  at MvvmCross.Droid.Support.V7.RecyclerView.MvxRecyclerAdapter.OnCreateViewHolder (Android.Views.ViewGroup parent, System.Int32 viewType) [0x00018] in C:\projects\mvvmcross\MvvmCross-AndroidSupport\MvvmCross.Droid.Support.V7.RecyclerView\MvxRecyclerAdapter.cs:157 
  at Android.Support.V7.Widget.RecyclerView+Adapter.n_OnCreateViewHolder_Landroid_view_ViewGroup_I (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_parent, System.Int32 viewType) [0x0000f] in <1c3b474ba0bc45cba6ec33f4b4b4b99a>:0 
  at (wrapper dynamic-method) System.Object:1e886010-d244-4f54-a83a-39d099b08368 (intptr,intptr,intptr,int)
  --- End of managed Android.Content.Res.Resources+NotFoundException stack trace ---

```

### Configuration :wrench:

**Version:** 5.6.3

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [ ] :monkey: Xamarin.Forms

### PR DESCRIPTION
**Pull request by orzech85**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
bug fix

### :arrow_heading_down: What is the current behavior?

it's not consistent between RecyclerView and Spinner and ListView

### :new: What is the new behavior (if this is a feature change)?

now is consistent

### :boom: Does this PR introduce a breaking change?
IMvxRecyclerViewHolder is extended by Id property

### :bug: Recommendations for testing
CollectionView in Playground

### :memo: Links to relevant issues/docs
fixes #2544

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [ ] Rebased onto current develop

### CODE REVIEW

File: MvvmCross.Android.Support/V7.RecyclerView/MvxRecyclerAdapter.cs [hunk 1]
```diff
@@ -167,12 +168,24 @@ public override int GetItemViewType(int position)
         protected virtual View InflateViewForHolder(ViewGroup parent, int viewType, IMvxAndroidBindingContext bindingContext)
         {
             var layoutId = ItemTemplateSelector.GetItemLayoutId(viewType);
-            return bindingContext.BindingInflate(layoutId, parent, false);
+            var view = bindingContext.BindingInflate(layoutId, parent, false);
+
+            if (layoutId == global::Android.Resource.Layout.SimpleListItem1) {
```
Comment (Cheesebaron) [inline MvvmCross.Android.Support/V7.RecyclerView/MvxRecyclerAdapter.cs]: braces should go on new line

File: MvvmCross.Android.Support/V7.RecyclerView/MvxRecyclerAdapter.cs [hunk 2]
```diff
@@ -167,12 +168,24 @@ public override int GetItemViewType(int position)
...
+                view.Tag = new SimpleItemTag();
+            }
+
+            return view;
+        }
+
+        public class SimpleItemTag : Object
```
Comment (Cheesebaron) [inline MvvmCross.Android.Support/V7.RecyclerView/MvxRecyclerAdapter.cs]: Do we really need this? Couldn't it be `global::Android.Resource.Layout.SimpleListItem1` for the tag you use for the view?

File: MvvmCross.Android.Support/V7.RecyclerView/MvxRecyclerAdapter.cs [hunk 3]
```diff
@@ -167,12 +168,24 @@ public override int GetItemViewType(int position)
...
+        {
         }
 
         public override void OnBindViewHolder(Android.Support.V7.Widget.RecyclerView.ViewHolder holder, int position)
         {
             var dataContext = GetItem(position);
+            if (holder.ItemView.Tag is SimpleItemTag)
+                (holder.ItemView as TextView).Text = dataContext.ToString();
```
Comment (Cheesebaron) [inline MvvmCross.Android.Support/V7.RecyclerView/MvxRecyclerAdapter.cs]: Check for null on `dataContext`

File: MvvmCross.Android.Support/V7.RecyclerView/MvxRecyclerAdapter.cs (hunk 2 above)
Comment (orzech85) [reply MvvmCross.Android.Support/V7.RecyclerView/MvxRecyclerAdapter.cs]: I think it is easier to check the type in `OnBindViewHolder` then cast `Tag` it to `int`. We are never be sure what is in the `Tag`. `OnBindViewHolder` is called relatively often then I would like to do as less as possible checks.

Comment (Cheesebaron) [reply MvvmCross.Android.Support/V7.RecyclerView/MvxRecyclerAdapter.cs]: Technically we could also expand the `IMvxRecyclerViewHolder` interface to contain the id used for the view. Then we only need to check that and don't need to create any Java objects. We cast the holder every time anyways. I think that would be the better solution.

**Pull request merged**